python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -710,6 +711,30 @@ def do_evaluate(
                 flags=plc.strings.regex_flags.RegexFlags.DEFAULT,
             )
             return Column(plc.strings.contains.contains_re(column.obj, prog))
+        elif self.name == pl_expr.StringFunction.Slice:
+            child, expr_start, expr_length = self.children
+            assert isinstance(expr_start, Literal)
+            assert isinstance(expr_length, Literal)
+
+            column = child.evaluate(df, context=context, mapping=mapping)
+
+            # libcudf slices via [start,stop).
+            # polars slices with offset + length where start == offset
+            # stop = start + length. Do this math on the host
+            stop = Literal(
+                expr_length.dtype,
+                pa.scalar(
+                    expr_start.value.as_py() + expr_length.value.as_py(),
+                    type=pa.int32(),
+                ),
+            ).evaluate(df, context=context, mapping=mapping)
+            start = expr_start.evaluate(df, context=context, mapping=mapping)
```

Comment:
Since you know you're making scalars, you can just do `plc.interop.from_arrow` on the pyarrow scalar. for the one you computed.